src/lib/Siteaccess/NonAdminSiteaccessResolver.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Siteaccess;
+
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\Core\Base\Exceptions\BadStateException;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+
+class NonAdminSiteaccessResolver
```

Comment:
We have `NonAdminSiteaccessResolver` and `SiteaccessResolver`, both share few method signatures. I guess it would be best if they implemented same interface.